misc/cron/archive.php

```diff
@@ -59,41 +52,9 @@
     array_unshift($_SERVER['argv'], $script);
 
     $console->run();
-} else { // if running via web request, use CronArchive directly
-
-    if (Piwik\Common::isPhpCliMode()) {
-        // We can run the archive in CLI with `php-cgi` so we have to configure the container/logger
-        // just like for CLI
-        $environment = new \Piwik\Application\Environment('cli');
-        $environment->init();
-
-        /** @var ConsoleHandler $consoleLogHandler */
-        $consoleLogHandler = StaticContainer::get('Symfony\Bridge\Monolog\Handler\ConsoleHandler');
-        $consoleLogHandler->setOutput(new ConsoleOutput(OutputInterface::VERBOSITY_VERBOSE));
-    } else {
-        // HTTP request: logs needs to be dumped in the HTTP response (on top of existing log destinations)
-        $environment = new \Piwik\Application\Environment(null);
-        $environment->init();
-
-        /** @var \Monolog\Logger $logger */
-        $logger = StaticContainer::get('Psr\Log\LoggerInterface');
-        $handler = new StreamHandler('php://output', Logger::INFO);
-        $handler->setFormatter(StaticContainer::get('Piwik\Plugins\Monolog\Formatter\LineMessageFormatter'));
-        $logger->pushHandler($handler);
-    }
-
-    $archiver = new Piwik\CronArchive();
-
-    if (!Piwik\Common::isPhpCliMode()) {
-        $token_auth = Piwik\Common::getRequestVar('token_auth', '', 'string');
-
-        if (!$archiver->isTokenAuthSuperUserToken($token_auth)) {
-            die('<b>You must specify the Super User token_auth as a parameter to this script, eg. <code>?token_auth=XYZ</code> if you wish to run this script through the browser. </b><br>
-                    However it is recommended to run it <a href="http://piwik.org/docs/setup-auto-archiving/">via cron in the command line</a>, since it can take a long time to run.<br/>
-                    In a shell, execute for example the following to trigger archiving on the local Piwik server:<br/>
-                    <code>$ /path/to/php /path/to/piwik/console core:archive --url=http://your-website.org/path/to/piwik/</code>');
-        }
-    }
+} else { // if running via web request, use CoreAdminHome.runCronArchiving method
+    $_GET['module'] = 'API';
+    $_GET['method'] = 'CoreAdminHome.runCronArchiving';
 
-    $archiver->main();
+    require_once PIWIK_INCLUDE_PATH . "/index.php";
```

Comment:
> Ah ok, and so the CoreAdminHome.runCronArchiving will be called by the archiver directly? It would make sense indeed.

By archiver do you mean archive.php? That would skip the authentication setup done by FrontController. I think ideally, w/ application encapsulation, we could do:

```
$webApplication = new Piwik\WebApplication();
$webApplication->dispatch(...);
```

but that's not possible yet. I could use FrontController instead of setting `$_GET` though.